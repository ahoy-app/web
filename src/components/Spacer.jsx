import React from 'react'
import PropTypes from 'prop-types'

const Spacer = ({ top = 0, bottom = 0, left = 0, right = 0, children }) => (
  <div style={{ padding: `${top}px ${right}px ${bottom}px ${left}px` }}>
    {children}
  </div>
)

Spacer.propTypes = {
  top: PropTypes.number,
  bottom: PropTypes.number,
  left: PropTypes.number,
  right: PropTypes.number,
  children: PropTypes.object,
}

export default Spacer
