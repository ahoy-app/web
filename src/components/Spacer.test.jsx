import React from 'react'
import { shallow } from 'enzyme'

import Spacer from './Spacer'

describe('Spacer', () => {
  it('should render correctly', () => {
    const component = shallow(<Spacer />)
    expect(component).toMatchSnapshot()
  })
})
