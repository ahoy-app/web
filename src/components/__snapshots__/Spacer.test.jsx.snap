// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should render correctly 1`] = `
<div
  style={
    Object {
      "padding": "0px 0px 0px 0px",
    }
  }
/>
`;
